src/App.js

```diff
@@ -4,25 +4,37 @@ import TodoList from "./TodoList";
 import AddTodoForm from "./AddTodoForm";
 
 
-const useSemiPersistentState = () =>{
+function App() {
 
-    const [todoList, setTodoList] = React.useState(
-        JSON.parse(localStorage.getItem('savedTodoList') || '[]' )
-    );
+    const [todoList, setTodoList] = React.useState([]);
+
+    const [isLoading, setIsLoading] = React.useState(true);
 
 
     React.useEffect(() => {
-        localStorage.setItem('savedTodoList', JSON.stringify(todoList));
-    }, [todoList]);
 
-    return [todoList, setTodoList];
-}
+        new Promise((resolve, reject) =>
+            setTimeout(
+                () => resolve({data: {todoList:
+                            JSON.parse(localStorage.getItem('savedTodoList'))
+                }}),
+                2000
+            ),
+        ).then((result) => {
+            setTodoList(JSON.parse(localStorage.getItem('savedTodoList')))
+            setIsLoading(false);
+        }
+        );
 
+    }, []);
 
 
-function App() {
+    React.useEffect(() => {
+        if(isLoading == false) {
+            localStorage.setItem('savedTodoList', JSON.stringify(todoList));
+        }
+    }, [todoList]);
```

Comment:
you should add `isLoading` as a dependency here so that this effect will be run when `isLoading` changes